doc/modules/model_evaluation.rst

```diff
@@ -2711,6 +2711,45 @@ Here are some usage examples of the :func:`d2_absolute_error_score` function::
   >>> d2_absolute_error_score(y_true, y_pred)
   0.0
 
+Visual evaluation of regression models
+--------------------------------------
+
+The :class:`~sklearn.metrics.PredictionErrorDisplay` displays allows to
+visually inspect the quality of regression models. The quality of a regression
+model can be assessed using two different plots as shown below:
+
+.. image:: ../auto_examples/model_selection/images/sphx_glr_plot_cv_predict_001.png
+   :target: ../auto_examples/model_selection/plot_cv_predict.html
+   :scale: 75
+   :align: center
+
+The plot on the left shows the predicted values vs. the actual values. A
+perfect regression model would display data point on the diagonal defined by
+predicted = actual values. The further away from this optimal line, the larger
+the error of the model. Note that this analysis is only meaningful when the
+predicted values is the expectation. This plot cannot be used when the model
+predicts a specific quantile, e.g. using
+:class:`~sklearn.linear_model.QuantileRegressor`.
+
+The right-hand side plot shows the residuals, i.e. the difference between the
+actual values and the predicted values, vs. the predicted values. This type of
+plot give an indication of the `homoscedasticity
+<https://en.wikipedia.org/wiki/Homoscedasticity_and_heteroscedasticity>`_ of
+the model. A perfect regression model will display data points uniformly
+distributed around the horizontal axis. If the distribution is not uniform,
+then it indicates us that there is a relationship between the value of the
+target and the residuals, e.g. the error increases when the true target
+increases.
```

Comment:
There is no reason for the "perfect model" to display homoschedastic residuals whenever the irreducible error is non-zero. If the irreducible error is null, then
the perfect model has zero residuals and the residuals homoschedastic as a consequence
but the homoschedasticity of a deterministic "random" variable is rather pointless.

Also the horizontal axis is not a density function so it does not make sense to speak about a uniform distribution when reading this plot... The distributions of residuals is rather to be "read vertically" for a given narrow, horizontal slice of `y_pred`.

Let' me suggest the following instead"
 
```suggestion
The right-hand side plot shows the residuals, i.e. the difference between the
actual values and the predicted values, vs. the predicted values.

This plot makes it easier to visualize if the residuals follow
and `homoscedastic or heteroschedastic
<https://en.wikipedia.org/wiki/Homoscedasticity_and_heteroscedasticity>`_
distribution. In particular, if the true distribution of `y|X` is Poisson
or Gamma distributed, it is expected that the variance of the residuals
of the optimal model to grow with the predicted value of `E[y|X]` (either linearly
for Poisson or quadratically for Gamma).

We can also use this plot to check if the residuals are Gaussian distributed
with a constant variance (homeschedastic residuals) which the assumption made
when fitting linear least squares regression model (see
:class:`sklearn.linear_mnodel.LinearRegression` and
:class:`sklearn.linear_mnodel.Ridge`). If this is not the case, and in particular if
the residual plot some banana-shaped structure, this is a hint that the model is
likely mis-specified and that non-linear feature engineering or switching to a
non-linear regression model might be useful.
```

Reply:
For reference, here is more information on how to interpret those kinds of plots:

https://www.qualtrics.com/support/stats-iq/analyses/regression-guides/interpreting-residual-plots-improve-regression/

I think it would be great to write some kind of advanced tutorial to diagnose various kinds of model misspecification problems for linear regression (penalized or not) but this should be done in a dedicated PR.